### ISSUE TITLE
jsx-pascal-case 

### ISSUE THREAD

**Issue by bowencool** [vuejs/eslint-plugin-vue#1671]
**Please describe what the rule should do:**
https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-pascal-case.md

**What category should the rule belong to?**

[x] Enforces code style (layout)
[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
<!-- bad -->
<Test_component />
<TEST_COMPONENT />
<testComponent />
<!-- good -->
<TestComponent>
  <div />
</TestComponent>
```

**Additional context**

https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-pascal-case.md
